### PR TITLE
Document what `roots_of_unity` computes

### PR DESCRIPTION
**Pull request by Pratyush**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (main)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: poly/src/domain/radix2/fft.rs [hunk 1]
```diff
@@ -60,21 +60,20 @@ impl<F: FftField> Radix2EvaluationDomain<F> {
         ark_std::cfg_iter_mut!(x_s).for_each(|val| *val *= self.size_inv);
     }
 
+    /// Computes the first `self.size / 2` roots of unity.
```
Comment (ValarDragon) [inline poly/src/domain/radix2/fft.rs:63]:
```suggestion
    /// Computes the first `self.size / 2` roots of unity for the entire domain.
    /// e.g. for the domain [1, g, g^2, ..., g^{n - 1}], it computes
    // [1, g, g^2, ..., g^{(n/2) - 1}]
```

**Comment by ValarDragon** [comment]
Don't we need to change the serial case to half its size?

**Comment by Pratyush** [comment]
That's what the PR does [here](https://github.com/arkworks-rs/algebra/blob/a55748e92ba03b121d4b172f0a3d315b72817add/poly/src/domain/radix2/fft.rs#L68)

**Comment by ValarDragon** [comment]
Oh lol, I somehow missed that, and only saw the base case's of the parallel case having its size changed.

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**